internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,604 @@
+			newSecretEnvVar("DB_NAME", "db.name", databaseSecretName),
+			newSecretEnvVar("DB_PASS", "db.password", databaseSecretName),
+			newSecretEnvVar("DB_PORT", "db.port", databaseSecretName),
+			newSecretEnvVar("DB_USER", "db.user", databaseSecretName),
+
+			// image overrides
+			newEnvVar("AGENT_DOCKER_IMAGE", AgentImage()),
+			newEnvVar("CONTROLLER_IMAGE", ControllerImage()),
+			newEnvVar("INSTALLER_IMAGE", InstallerImage()),
+			newEnvVar("SELF_VERSION", ServiceImage()),
+
+			{
+				Name: "NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+		},
+		EnvFrom: []corev1.EnvFromSource{
+			{
+				ConfigMapRef: &corev1.ConfigMapEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: assistedServiceConfigMapName,
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "bucket-filesystem",
+				MountPath: "/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(2000*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+			InitialDelaySeconds: 3,
+			PeriodSeconds:       10,
+			TimeoutSeconds:      3,
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{
+					Port: intstr.FromInt(int(servicePort)),
+				},
+			},
+		},
+	}
+
+	postgresContainer := corev1.Container{
+		Name:            databaseContainerName,
+		Image:           DatabaseImage(),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          databaseName,
+				ContainerPort: databasePort,
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		Env: []corev1.EnvVar{
+			newSecretEnvVar("POSTGRESQL_DATABASE", "db.name", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_USER", "db.user", databaseSecretName),
+			newSecretEnvVar("POSTGRESQL_PASSWORD", "db.password", databaseSecretName),
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "postgresdb",
+				MountPath: "/var/lib/pgsql/data",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(200, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(500*1024*1024, resource.BinarySI),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+				corev1.ResourceMemory: *resource.NewQuantity(400*1024*1024, resource.BinarySI),
+			},
+		},
```

Comment:
Should these be configurable?

Reply:
Answering in https://github.com/openshift/assisted-service/pull/1422#discussion_r607129857